src/library/scala/collection/IndexedSeq.scala

```diff
@@ -25,6 +25,28 @@ trait IndexedSeq[+A] extends Seq[A]
   override protected[this] def stringPrefix: String = "IndexedSeq"
 
   override def iterableFactory: SeqFactory[IndexedSeq] = IndexedSeq
+
+  override def foldLeft[B](z: B)(f: (B, A) => B): B = {
```

Comment:
Just a small comment.
```
  override def foldLeft[B](z: B)(f: (B, A) => B): B = {
    val LEN = length
    @tailrec def inner(acc: B, n: Int): B = n match {
      case LEN => acc
      case _ => inner(f(acc, apply(n)), n+1)
    }
    inner(z, 0)
  }
```
I don't know if this is a more idiomatic way to do `foldLeft`. The benefit is, well, getting rid of `var`. In this implementation, it surely is harmless. What is your opinion?

Reply:
Hey @cvogt. From my perspective, performance is my main priority for this PR. If two impls are equally performant I'd prefer the simpler / more readable one. Personally I find a while-loop considerably more readable for these two specific cases :)

Reply:
IMO - the answer is to do a JMH benchmark and see which is better.
It they both perform the same then we can work out which is the prettier code, and generates more idiomatic bytecode.

Reply:
I did a benchmark for the two implementations. https://gist.github.com/contrun/0e02b1ee19aa61dd33450dbaca7a6454. It seems the tailrec version is a little fast. I am not familiar with jmh. I don't know if the benchmark is representative.

Reply:
Actually doing a benchmark is great!  In this case, the two numbers are not statistically different (see sizes of errors; can do a two-sample T test or something).  So we should conclude from this benchmark that there is no evidence for using one or the other, but we probably can't tell differences of 10% or less given the parameters used.

FWIW, I always use while loops in simple cases like this.  The cognitive overhead of writing inner methods (keeping parameter order straight, etc.) is, for me at least, greater than the cognitive overhead of keeping track of one local mutable variable.

Reply:
The while loop one is more simple.

Reply:
Benchmark looks good. A bit too much noise in the first one - did you have something else running?
To compare like with like on the benchmark you should capture length I think
```
    var b = z
    var i = 0
    while (i < length) {
      val a = apply(i)
...
```
should be
```
    val len = length
    var b = z
    var i = 0
    while (i < len) {
      val a = apply(i)
...
```
That should avoid a call to length in each iteration

FWIW I prefer the while loop, as the code that is generated is more like java bytecode, so more likely that the JVM will recognise the pattern and optimise it. E.g. capturing the length allows the JVM to unroll the loop